File: crossecore-java/src/java_/ArrayLengthAccessBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ArrayLengthAccessBase 
extends ExpressionImpl implements ArrayLengthAccess
{
	private Expression _array;
	public Expression getArray()
	{
		
			return _array;
		
		
	}
	public void setArray(Expression value){
		if (value != _array) {
			NotificationChain msgs = null;
			if (_array != null){
				msgs = ((InternalEObject)_array).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY, null, msgs);
			}
			msgs = basicSetArray(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.ARRAYLENGTHACCESS;
	}
	
	
	public NotificationChain basicSetArray(Expression newobj, NotificationChain msgs) {
		Expression oldobj = _array;
		_array = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY:
				return getArray();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY:
				setArray((Expression) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.ARRAYLENGTHACCESS_ARRAY:
				return _array != null; //single != null;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ArrayLengthAccessBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {Expression} from "./Expression";
import {ArrayLengthAccess} from "./ArrayLengthAccess";
		
			export class ArrayLengthAccessBase
			extends ExpressionImpl
			implements ArrayLengthAccess
			{
				private _array:Expression = null;
				get array():Expression{
				
					return this._array;
				}
				set array(value:Expression) {
					if (value != this._array) {
						let msgs:NotificationChain = null;
						if (this._array != null){
							msgs = (this._array).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY, /*null*/ null, msgs);
						}
						msgs = this.basicSetArray(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ArrayLengthAccessBase.eStaticClass;
				}
			
			
				public basicSetArray(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._array;
					this._array = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.ARRAYLENGTHACCESS_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.ARRAYLENGTHACCESS_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.ARRAYLENGTHACCESS_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY:
							return this.array;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.ARRAYLENGTHACCESS_ARRAY:
							this.array = <Expression> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
